client/packages/system/src/Immunisation/VaccineCourseEditModal/VaccineCourseEditModal.tsx

```diff
@@ -0,0 +1,289 @@
+      labelProps={{
+        sx: {
+          fontSize: '16px',
+          paddingRight: 2,
+          textAlign: 'right',
+        },
+      }}
+      Input={
+        <Box sx={{}} flex={1}>
+          {children}
+        </Box>
+      }
+    />
+  </Box>
+);
+
+interface VaccineCourseEditModalProps {
+  vaccineCourse: VaccineCourseFragment | null;
+  isOpen: boolean;
+  onClose: () => void;
+  programId: string | undefined;
+  mode: ModalMode | null;
+}
+
+export const VaccineCourseEditModal: FC<VaccineCourseEditModalProps> = (
+  { vaccineCourse, isOpen, onClose, programId, mode }
+) => {
+  const t = useTranslation('coldchain');
+  const { success, error } = useNotification();
+  const {
+    draft,
+    update: { update },
+    create: { create },
+    updatePatch,
+    query: { isLoading },
+    isDirty,
+    setIsDirty,
+  } = useVaccineCourse(vaccineCourse?.id ?? undefined);
+  const { data: demographicData } = useDemographicIndicators();
+
+  const { Modal } = useDialog({ isOpen, onClose, disableBackdrop: true });
+  const height = useKeyboardHeightAdjustment(600);
+
+  const tryUpdateValue = (value: number | undefined) => {
+    if (value === undefined) {
+      return;
+    }
+    updatePatch({ doses: value });
+  };
+
+  // const updateSchedule = (value: number) => {
+  //   if (!value) {
+  //     return;
+  //   }
+  //   const scheduleSeed = (number: number): VaccineCourseScheduleNode => {
+  //     return {
+  //       __typename: 'VaccineCourseScheduleNode',
+  //       id: FnUtils.generateUUID(),
+  //       doseNumber: number,
+  //       label: '',
+  //     };
+  //   };
+  //   let rows = draft?.vaccineCourseSchedules ?? [];
+
+  //   if (rows.length === value) {
+  //     return;
+  //   } else if (value > rows.length) {
+  //     let toAdd = value - rows.length;
+  //     while (toAdd > 0) {
+  //       const number = value - toAdd + 1;
+  //       rows.push(scheduleSeed(number));
+  //       toAdd--;
+  //     }
+  //   } else {
+  //     rows = rows.slice(0, value);
+  //   }
+  //   updatePatch({ vaccineCourseSchedules: rows });
+  // };
+
+  // const updateDescription = (patch: RecordPatch<VaccineCourseScheduleNode>) => {
+  //   if (!patch) {
+  //     return;
+  //   }
+  //   const schedule = {
+  //     ...ArrayUtils.toObject(draft.vaccineCourseSchedules ?? [defaultRow]),
+  //     [patch.id]: patch,
+  //   };
+
+  //   const scheduleArray = Object.values(
+  //     schedule
+  //   ) as VaccineCourseScheduleNode[];
+  //   updatePatch({ vaccineCourseSchedules: scheduleArray });
+  // };
+
+  // const dosesColumns = useColumns<VaccineCourseScheduleNode>(
+  //   [
+  //     { key: 'doseNumber', label: 'label.dose-number' },
+  //     [descriptionColumn(t('label.new-row')), { setter: updateDescription }],
+  //   ],
+  //   {},
+  //   [draft]
+  // );
+
+  const options = useMemo(
+    () => getDemographicOptions(demographicData?.nodes ?? []),
+    [demographicData]
+  );
+
+  if (isLoading) {
```

Comment:
The `BasicSpinner` is shown in preference to the modal, so you get this displayed when loading:

<img width="437" alt="image" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/7cf98531-f38f-4437-a10c-bf21b5324a3c">

if you change to something like this 

```
  const modalContent = isLoading ? (
    <BasicSpinner />
  ) : (
    <Box display="flex" flex={1}>
      <Container>
        <Row label={t('label.immunisation-name')}>
          <BasicTextInput
            value={draft?.name ?? ''}
            fullWidth
            onChange={e => updatePatch({ name: e.target.value })}
          />
        </Row>
        <Row label={t('label.target-demographic')}>
          <Autocomplete
            isOptionEqualToValue={option =>
              option?.value === draft.demographicIndicatorId
            }
            onChange={(_e, selected) =>
              updatePatch({ demographicIndicatorId: selected?.value })
            }
            defaultValue={defaultValue}
            options={options}
          />
        </Row>
        <Row label={t('label.coverage-rate')}>
          <NumericTextInput
            value={draft?.coverageRate ?? 1}
            fullWidth
            onChange={value => updatePatch({ coverageRate: value })}
          />
        </Row>
        <Row label={t('label.wastage-rate')}>
          <NumericTextInput
            value={draft?.wastageRate ?? 1}
            fullWidth
            onChange={value => updatePatch({ wastageRate: value })}
          />
        </Row>
        <Row label={t('label.vaccine-items')}>
          <VaccineItemSelect draft={draft} onChange={updatePatch} />
        </Row>
        <Row label={t('label.calculate-demand')}>
          <Checkbox
            checked={draft?.isActive ?? true}
            onChange={e => updatePatch({ isActive: e.target.checked })}
          ></Checkbox>
        </Row>

        <Row label={t('label.number-of-doses')}>
          <NumericTextInput
            value={draft.doses}
            fullWidth
            onChange={tryUpdateValue}
          />
        </Row>
      </Container>
    </Box>
  );

  return (
    <Modal
      title={
        mode === ModalMode.Create
          ? t('heading.create-vaccine-course')
          : t('heading.edit-vaccine-course')
      }
      cancelButton={<DialogButton variant="cancel" onClick={onClose} />}
      okButton={
        <DialogButton
          disabled={!isDirty || !programId}
          variant="ok"
          onClick={save}
        />
      }
      height={height}
      width={700}
      slideAnimation={false}
    >
      {modalContent}
    </Modal>
  );
``` 

then you get 
 
<img width="987" alt="image" src="https://github.com/msupply-foundation/open-msupply/assets/9192912/4d435113-98f3-44ad-bc25-f221f8a1df3f">